File: src/main/java/com/github/pauldambra/moduluschecker/chain/checks/ModulusTenCheck.java
package com.github.pauldambra.moduluschecker.chain.checks;

import com.github.pauldambra.moduluschecker.ModulusCheckParams;
import com.github.pauldambra.moduluschecker.valacdosFile.WeightRow;

import java.util.function.Function;

public class ModulusTenCheck {

    public Boolean check(ModulusCheckParams params, Function<ModulusCheckParams, WeightRow> rowSelector) {
        WeightRow selectedRow = rowSelector.apply(params);

        int total = ModulusTotal.calculate(params.account, selectedRow.getWeights());

        return total % 10 == 0;
    }

}

File: src/main/java/com/github/pauldambra/moduluschecker/chain/checks/ModulusTenCheck.kt
package com.github.pauldambra.moduluschecker.chain.checks

import com.github.pauldambra.moduluschecker.account.BankAccount
import com.github.pauldambra.moduluschecker.valacdosFile.WeightRow

class ModulusTenCheck {

    fun check(weightRow: WeightRow, bankAccount: BankAccount): Boolean {

        val total = ModulusTotal.calculate(bankAccount, weightRow.weights)

        return total % 10 == 0
    }

}
